File: hexample-parent/adapter/src/main/java/at/willhaben/sp/hexample/adapter/output/database/DatabaseAdapter.java
package at.willhaben.sp.hexample.adapter.output.database;

import at.willhaben.sp.hexample.domain.jobsearch.ports.JobSearchDatabasePort;
import at.willhaben.sp.hexample.domain.model.JobOffer;
import at.willhaben.sp.hexample.domain.recommendation.ports.JobRecommendationDatabasePort;
import org.springframework.stereotype.Component;

import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.IntStream;

@Component
public class DatabaseAdapter implements JobSearchDatabasePort, JobRecommendationDatabasePort {

    private static List<JobOffer> jobOffersList;

    static {
        jobOffersList = IntStream.range(1, 11)
                .mapToObj(index -> {
                    JobOffer jobOffer = new JobOffer();
                    jobOffer.setTitle("Job " + index);
                    return jobOffer;
                })
                .collect(Collectors.toList());
    }

    @Override
    public List<JobOffer> searchForJobsWithTitle(String title) {
        return jobOffersList.stream()
                .filter(item -> title == null || title.equals(item.getTitle()))
                .collect(Collectors.toList());
    }

    @Override
    public List<JobOffer> getJobsForIdList(List<Integer> jobIds) {
        return jobIds.stream()
                .map(id -> jobOffersList.get(id))
                .collect(Collectors.toList());
    }

}

File: hexample-parent/adapter/src/main/kotlin/at/willhaben/sp/hexample/adapter/output/database/DatabaseAdapter.kt
package at.willhaben.sp.hexample.adapter.output.database

import at.willhaben.sp.hexample.domain.jobsearch.ports.JobSearchDatabasePort
import at.willhaben.sp.hexample.domain.model.JobItem
import at.willhaben.sp.hexample.domain.recommendation.ports.JobRecommendationDatabasePort
import org.springframework.stereotype.Repository
import java.util.stream.IntStream
import kotlin.streams.asSequence

@Repository
class DatabaseAdapter : JobSearchDatabasePort, JobRecommendationDatabasePort {
    private val jobItems = IntStream.range(1, 11)
        .asSequence()
        .map { JobItem(title = "Job $it") }
        .toList()

    override fun searchForJobsWithTitle(title: String?): List<JobItem> {
        return jobItems.filter { title == null || it.title == title }.toList()
    }

    override fun getJobsForIdList(jobIds: List<Int>): List<JobItem> {
        return jobIds.map { jobItems[it] }
    }
}
